mne/io/edf/edf.py

```diff
@@ -1046,6 +1046,18 @@ def _check_stim_channel(stim_channel, ch_names, sel):
     else:
         raise ValueError('Invalid stim_channel')
 
+    # Forbid the synthesis of stim channels from Annotations
+    # XXX: This is not how this should be done, but to check
+    #      the type of channel it is.
+    stim_name_ends_with_annotations = [ch for ch in valid_stim_ch_names
+                                       if ch.endswith('annotations')]
+    if len(stim_name_ends_with_annotations):
```

Comment:
When does this evaluate to `True`? If someone has a channel ending with "annotations"? If so, shouldn't this channel be treated as a stim channel (i.e. set its type to `stim` and don't synthesize)?

Reply:
This is true when someone passes `'EDF Annotations'` or `'BDF Annotations'`, in that case the user should take it out from the `stim_channel`. Therefore the `ValueError`.

Then, if an annotations channel is present in the file it gets loaded as annotations, and there is no stim channel with such name.

Reply:
However, this also includes other channels ending with "annotations", which is probably not what we want. I'd exclude only these 2 particular names.

Reply:
Again, the right thing to do would be to actually get a list of **TAL** channels when `_read_edf_header` and raise the error based on that. But at this point, I think that we can do that in a different PR (if we ever get to it).

But sure, we can check for those specific names only.